app/controllers/weather_controller.rb

```diff
@@ -0,0 +1,10 @@
+class WeatherController < ApplicationController
+  def temperature
```

Comment:
Here you are depending on an external service, is better to use asynchronous request, with this gem 'async_request'.
https://github.com/Wolox/async-requests